src/dials/algorithms/indexing/stills_indexer.py

```diff
@@ -819,6 +820,21 @@ def refine(self, experiments, reflections):
 
         return ref_experiments, reflections
 
+    def warn_if_setting_unused_params(self):
+        warning_message = (
+            "Warning: the value of indexing.refinement_protocol.{} has been"
+            " changed to {}, but this parameter is unused by stills indexer."
+        )
+        unused_refinement_protocol_defaults = {
+            "n_macro_cycles": 5,
+            "d_min_step": libtbx.Auto,
+            "d_min_final": None,
+        }
```

Comment:
If the defaults change in the future, they'd need to be changed here too.  Instead of hard-coding here, could you import the un-fetched params and check them against the user params?

Reply:
I happily would if I knew how 😄

Reply:
It would be something like:
```
from dials.algorithms.indexing.indexer import phil_scope as defaults_phil_scope
defaults = defaults_phil_scope.extract
if self.params.refinement_protocol != defaults.indexing.refinement_protocol:
  ...
```

Reply:
Makes sense. I was looking for defaults somewhere in the parsed phil scope, but this is straightforward. Will do.

Reply:
Fixed and tested, warnings are the same, but defaults are now pulled automatically:
```
Warning: the value of indexing.refinement_protocol.n_macro_cycles has been changed to 4, but this parameter is unused by stills indexer.
Warning: the value of indexing.refinement_protocol.d_min_step has been changed to 2.0, but this parameter is unused by stills indexer.
Warning: the value of indexing.refinement_protocol.d_min_final has been changed to 2.0, but this parameter is unused by stills indexer.
```